### PR TITLE
fix: do not bypass torch.distributed.launch for single-slot trials

### PR DESCRIPTION
**Pull request by azhou-determined**
## Description

we bypass torch.distributed.launch behavior by automatically detecting single-slot trials and just calling the training script. 

## Checklist
- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.
- [ ] If modifying `/webui/react/src/shared/` verify `make -C webui/react test-shared` passes.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by rb-determined-ai — COMMENTED** [review]
release note?

File: docs/release-notes/torch-distributed-single-slot.txt [hunk 1]
```diff
@@ -0,0 +1,8 @@
+:orphan:
+
+**Breaking Changes**
+-  Distributed training: previously, experiments launched with determined.launch.torch_distributed was wrongly skipping
```
Comment (rb-determined-ai) [inline docs/release-notes/torch-distributed-single-slot.txt:4]: experiments ... were

**Review by rb-determined-ai — APPROVED** [review]
